internal/landing/landing.go

```diff
@@ -8,22 +8,69 @@ package landing
 
 import (
 	"embed"
+	"fmt"
+	"html/template"
 	"io/fs"
 	"net/http"
+	"net/url"
+
+	"github.com/talos-systems/discovery-service/internal/state"
 )
 
 //go:embed "html/index.html"
 var static embed.FS
 
+//go:embed "templates/inspect.html.tmpl"
+var inspectTemplate []byte
+
+// ClusterInspectData represents all affiliate data asssociated with a cluster.
+type ClusterInspectData struct {
+	ClusterID  string
+	Affiliates []*state.AffiliateExport
+}
+
+// InspectHandler provides data to inspect a cluster.
+func InspectHandler(w http.ResponseWriter, req *http.Request, state *state.State) error {
+	clusterData := ClusterInspectData{}
+
+	if formData, err := url.ParseQuery(req.URL.RawQuery); err == nil {
+		clusterData.ClusterID = formData.Get("clusterID")
+		fetchedCluster := state.GetCluster(clusterData.ClusterID)
+		clusterData.Affiliates = fetchedCluster.List()
+	} else {
+		return err
+	}
+
+	inspectPage, err := template.New("inspect").Parse(string(inspectTemplate))
```

Comment:
I think we should parse the template once with `template.Must(template.Parse(...))` and store it in a global variable to avoid re-parsing it every time.